saleor/graphql/order/resolvers.py

```diff
@@ -31,10 +31,11 @@ def filter_orders(qs, info, created, status, query):
 
 def resolve_orders(info, created, status, query):
     user = info.context.user
-    if user.has_perm('order.manage_orders'):
-        qs = models.Order.objects.confirmed()
-    else:
-        qs = user.orders.confirmed()
+
+    if not user.has_perm('order.manage_orders'):
```

Comment:
I think we could as well wrap the order resolver with `@permission_required('order.manage_orders')` just as we do for any other query that requires particular permission. Usually, we do it in the `schema.py` file, in this case, it would have to be changed in the `OrderQueries` class.